sklearn/utils/__init__.py

```diff
@@ -636,3 +674,72 @@ def is_scalar_nan(x):
     # convert from numpy.bool_ to python bool to ensure that testing
     # is_scalar_nan(x) is True does not fail.
     return bool(isinstance(x, numbers.Real) and np.isnan(x))
+
+
+def _approximate_mode(class_counts, n_draws, rng):
```

Comment:
I wonder whether this deserves a clearer name

Reply:
`draw_from_class_counts`?

Reply:
It's not a random draw, since we actually want the mode. Sorry... not coming up w good names here either